scripts/entity/components/movement_components/base_movement_component.gd

```diff
@@ -0,0 +1,78 @@
+class_name BaseMovementComponent extends CharacterBody2D
+## Base movement class which allows the object to move when given input.
+## [BR]
+## Provides events for when movement events occur.
+
+## Event called when horizontal movement starts. Should be implemented in child class.
+@warning_ignore("UNUSED_SIGNAL")
+signal on_horizontal_movement_started(value: float )
+
+## Event called when horizontal movement is happening. Should be implemented in child class.
+@warning_ignore("UNUSED_SIGNAL")
+signal on_horizontal_movement(value: float )
+
+## Event called when horizontal movement ends. Should be implemented in child class.
+@warning_ignore("UNUSED_SIGNAL")
+signal on_horizontal_movement_cancelled(value: float )
+
+## Event called when a jump starts. Should be implemented in child class.
+@warning_ignore("UNUSED_SIGNAL")
+signal on_jump_started()
+
+## Event called when the entity becomes grounded after not being grounded.
+signal on_grounded_start()
+
+## Event called when the entity becomes airborne.
+signal on_airborne_start()
+
+## If the entity was grounded in the last physics frame.
+var _was_grounded: bool
+
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	push_error('Abstract Class Instaniation Error: %s' % [name])
+	assert (false, 'Abstract Class Instaniation Error: %s' % [name])
+
+## Determines whether or not the entity is grounded or airborne this frame and
+## saves the current state for use next frame.
+@warning_ignore("UNUSED_PARAMETER")
+func _physics_process(delta: float) -> void:
```

Comment:
if you put _delta then warning isn't sent by godot

Reply:
changed